src/fsharp/service/ServiceStructure.fs

```diff
@@ -610,14 +610,14 @@ module Structure =
                 parseAttributes attrs
             | _ -> ()
 
-        let parseModuleOrNamespace isScript (SynModuleOrNamespace (longId,_,isModule,decls,_,attribs,_,r)) =
+        let parseModuleOrNamespace (SynModuleOrNamespace (longId,_,kind,decls,_,attribs,_,r)) =
             parseAttributes attribs
             let idRange = longIdentRange longId
             let fullrange = Range.startToEnd idRange r  
             let collapse = Range.endToEnd idRange r 
         
             // do not return range for top level implicit module in scripts
-            if isModule && not isScript then
+            if kind <> AnonModule then
```

Comment:
The check should probably be `kind = NamedModule` since we don't want to include namespaces?

I think you might be right that we should have the same behavior for anon modules in scripts and non-scripts.

Reply:
I don't mind having an option to fold a namespace but it would require fixing folding ranges in Structure provider which is better to have in a separate PR. Thanks, I'll replace the check.
<img width="334" alt="screenshot 2018-12-19 at 19 13 18" src="https://user-images.githubusercontent.com/3923587/50232567-57051380-03c2-11e9-895b-1fdc85fc51ce.png">
<img width="452" alt="screenshot 2018-12-19 at 19 13 25" src="https://user-images.githubusercontent.com/3923587/50232568-57051380-03c2-11e9-9746-533a631e7e40.png">